docs/code-review.md

```diff
@@ -0,0 +1,97 @@
+
+Comments should be used to give praise for good solutions and to point out potential improvements. They should be not be used to criticise your colleagues or make strongly opinionated statements. Always be mindful of your tone, considering how others might perceive your comments, and be explicit about when a comment is non-blocking or unimportant.
+
+### Be constructive
+Don't use harsh language or criticise without making constructive suggestions.
+Do suggest alternative approaches and explain your reasoning.
+
+### Be specific
+Don't make vague statements about the changes as a whole.
+Do point out specific issues and offer specific ideas for how the changes can be improved.
+
+### Avoid strong opinions
+Don't make strong, opinionated statements or dictate specific changes.
+Do ask open-ended questions and keep an open mind about alternative solutions and reasoning that you may not have thought of.
+
+## Scope of a code review
+
+Code reviews should focus on what is being changed and whether the change is appropriate.
+
+Expanding the scope of a pull request at the review stage is not acceptable. It is generally more valuable to swiftly conclude a piece of work that the team has prioritised than to opportunistically seek addition changes, such as refactoring related code. That said, it can be useful to comment on refactoring opportunities without blocking the pull request. It may be that the author has some spare time they can use to make those changes, or can incorporate them with their next piece of work.
+
+Examples of things to look for and comment on in a code review:
+
+### What has changed
+Are the changes focussed on a specific issue, referenced in the pull request description? If the changes go beyond the intended scope, should they be broken up to make the code review more managable? If the code review is still a managable size, consider making a non-blocking comment to remind the author that they could control the scope of future pull requests more carefully.
+
+### Maintainability
+Is all new code extensible and easy for other developers to understand? Does it follow common design patterns? Look for unnecessary complexity and remember that this is subjective so take care not to be overly critical or opinionated when commenting on maintainability. Try to make specific suggestions for improvement, rather than simply stating a problem, but use open-ended language to encourage discussion.
+
+### Duplication
+Is there duplication within new code or between new and existing code? Could an existing abstraction be reused or should a new abstraction be created? Consider using non-blocking comments when proposing abstractions so they can be addressed separately as technical debt, rather than extending the scope of the pull request.
```

Comment:
Yes, a lot of that depends on the context of the change being performed. If it's the job of the PR to address something, add a new piece of code or extend existing functionality, then requesting a change to include more abstraction or a different interface would be appropriate.
If the change is a small modification, or "temporary" work on the path to something else, then sometimes abstraction can be unwanted.
As we're trying to keep the scope of PRs small, I think the suggestion to add something to the product backlog is a good one. Either it is a priority, it's debt and we need to keep it under control, so the issue will be dealt with as such. If it's not a priority then addressing it in the current PR is inappropriate anyway.
You're right that if this does happen often, then we need to collaborate more to consider the implementation of something before starting. Unless you're doing something new, you frequently know in advance what sort of approach you can take to fix a problem. Stories are small enough that unless the feature that you're working on is very small, the issue that you're fixing will just be a stepping stone on the way to a larger implementation. That will either have further items in the backlog already, or will need them adding as part of this PR.
So trying to solve the entire problem space in one go means that the story is too large and needs to be broken up.

Reply:
Really good point. How about this instead?
```
Is there duplication within new code or between new and existing code? Could an existing abstraction be reused or should a new abstraction be created? When proposing an abstraction, consider using a non-blocking comment to open a dialogue about what the abstraction should be and whether it needs addressing now or as a separate piece of work.

Identifying useful abstractions at the review stage may indicate that there wasn't enough collaboration before coding began. See this as a trigger to review the team's ways of working, rather than blocking a pull request unnecessarily.
```

Reply:
Looks good

Reply:
I think that's a good addition.

Reply:
Thanks, I've pushed that.

Reply:
That was the only point that stood out as being contentious, the new wording is better. Loks good to me.